C7Engine/MapUnitExtensions.cs

```diff
@@ -400,7 +400,16 @@ public static void disband(this MapUnit unit)
 
 	public static bool canBuildCity(this MapUnit unit)
 	{
-		return unit.unitType.actions.Contains("buildCity") && unit.location.IsAllowCities();
+		if (!unit.unitType.actions.Contains("buildCity")) {
+			return false;
+		}
+		if (unit.location.HasCity || !unit.location.IsAllowCities()) {
+			return false;
+		}
+		if (unit.location.neighbors.Values.Count(tile => tile.HasCity) > 0) {
```

Comment:
I suggest to use Any instead of Count.

Reply:
Good idea.  I wound up taking the idea from #337 and using All with !HasCity, IMO it's slightly easier to read that way.